### PR TITLE
Improve agent execution modes

### PR DESCRIPTION
**Pull request by sdvendramini**
|Related issue|
|---|
|https://github.com/wazuh/wazuh-agent/issues/288|

This PR aims to improve how the agent runs in different environments.

### CODE REVIEW

File: src/agent/src/unix_daemon.cpp [hunk 1]
```diff
@@ -59,77 +60,85 @@ namespace unix_daemon
         }
     }
 
-    bool PIDFileHandler::writePIDFile() const
+    bool LockFileHandler::createLockFile() const
     {
-        const std::string path = fmt::format("{}/{}", GetExecutablePath(), PID_PATH);
+        const std::string path = fmt::format("{}/{}", GetExecutablePath(), LOCK_PATH);
         createDirectory(path);
```
Comment (TomasTurina) [inline src/agent/src/unix_daemon.cpp:66]: Validate the return value.

**Review by TomasTurina — COMMENTED** [review]
Remember to update the `BUILD.md` file.

File: src/agent/src/process_options_unix.cpp [hunk 2]
```diff
@@ -11,33 +11,26 @@
...
 
-void RestartAgent(const std::string& configFile)
+void RestartAgent([[maybe_unused]] const std::string& configFile)
 {
-    StopAgent();
-
-    std::this_thread::sleep_for(std::chrono::seconds(1)); // NOLINT
-
-    StartAgent(configFile);
+    std::cout << "not implemented on Unix.\n";
```
[the remark this answers is not in the document]
Comment (sdvendramini) [reply src/agent/src/process_options_unix.cpp]: Done!

File: src/agent/src/process_options_unix.cpp [hunk 3]
```diff
@@ -47,24 +40,7 @@ void StatusAgent()
...
-    if (!kill(pid, SIGTERM))
-    {
-        LogInfo("Wazuh-agent terminated");
-    }
-    else
-    {
-        kill(pid, SIGKILL);
-        LogInfo("Wazuh-agent killed");
-    }
+    std::cout << "not implemented on Unix.\n";
```
[the remark this answers is not in the document]
Comment (sdvendramini) [reply src/agent/src/process_options_unix.cpp]: Done!

File: src/agent/src/unix_daemon.cpp (hunk 1 above)
Comment (sdvendramini) [reply src/agent/src/unix_daemon.cpp]: Done!

File: src/agent/src/process_options_unix.cpp [hunk 4]
```diff
@@ -74,14 +50,10 @@ void PrintHelp()
     std::cout << "Usage: wazuh-agent [options]\n";
     std::cout << "\n";
     std::cout << "Options:\n";
-    std::cout << "     " << OPT_RUN << "                Start wazuh-agent\n";
-    std::cout << "     " << OPT_START << "              Start wazuh-agent daemon\n";
```
[the remark this answers is not in the document]
Comment (sdvendramini) [reply src/agent/src/process_options_unix.cpp:78]: Fixed!

File: src/agent/src/unix_daemon.cpp [hunk 5]
```diff
@@ -59,77 +60,88 @@ namespace unix_daemon
...
-        const std::string filename = fmt::format("{}/{}/wazuh-agent.pid", GetExecutablePath(), PID_PATH);
-        std::ofstream file(filename);
-
-        if (!file.is_open())
+        const std::string path = fmt::format("{}/{}", GetExecutablePath(), LOCK_PATH);
+        if (!createDirectory(path))
         {
-            LogCritical("Unable to write PID file: {}. Error: {} ({})", filename.c_str(), errno, std::strerror(errno));
-            return false;
+            LogError("Unable to create lock directory: {}", path);
```
Comment (TomasTurina) [inline src/agent/src/unix_daemon.cpp:68]:
```suggestion
            LogError("Unable to create lock directory: {}", path);
            return false;
```

Comment (sdvendramini) [reply src/agent/src/unix_daemon.cpp:68]: Done!

File: src/agent/src/main.cpp [hunk 6]
```diff
@@ -52,13 +52,13 @@ int main(int argc, char* argv[])
         {
             SetDispatcherThread();
         }
-        else if (cmdParser.OptionExists(OPT_RUN) || cmdParser.OptionExists(OPT_START))
+        else if (cmdParser.OptionExists(OPT_HELP))
         {
-            StartAgent(configFile);
+            PrintHelp();
         }
         else
```
[the remark this answers is not in the document]
Comment (sdvendramini) [reply src/agent/src/main.cpp:60]: Done!

File: src/agent/include/cmd_ln_parser.hpp [hunk 7]
```diff
@@ -13,10 +14,15 @@ class CommandlineParser
     /// @brief Constructor
     /// @param argc The number of command line arguments
     /// @param argv The command line arguments
-    CommandlineParser(int argc, char** argv)
+    CommandlineParser(int argc, char** argv, const std::vector<std::string> options)
```
Comment (TomasTurina) [inline src/agent/include/cmd_ln_parser.hpp:17]: Missing new parameter in the description.

Comment (TomasTurina) [inline src/agent/include/cmd_ln_parser.hpp:17]:
```suggestion
    CommandlineParser(int argc, char** argv, const std::vector<std::string>& options)
```

File: src/agent/src/unix_daemon.hpp [hunk 8]
```diff
@@ -6,59 +6,48 @@
...
-        /// @param relativePath The relative path for the PID file
+        /// @brief Creates the directory path for the lock file
+        /// @param relativePath The relative path for the lock file
         /// @return True if the directory is created, false otherwise
-        bool createDirectory(const std::string_view relativePath) const;
-
-        /// @brief Writes the current process ID to the PID file
-        /// @return True if the PID file is written, false otherwise
-        bool writePIDFile() const;
+        bool createDirectory(const std::string& relativePath) const;
```
Comment (TomasTurina) [inline src/agent/src/unix_daemon.hpp:33]:
```suggestion
        bool createDirectory(const std::string& path) const;
```

File: src/agent/src/unix_daemon.hpp [hunk 9]
```diff
@@ -6,59 +6,48 @@
...
-        static pid_t ReadPIDFromFile();
+        /// @brief Removes the lock file
+        /// @return True if the file is removed, false otherwise
+        bool removeLockFile() const;
 
     private:
-        /// @brief Creates the directory path for the PID file
-        /// @param relativePath The relative path for the PID file
+        /// @brief Creates the directory path for the lock file
+        /// @param relativePath The relative path for the lock file
```
Comment (TomasTurina) [inline src/agent/src/unix_daemon.hpp:31]:
```suggestion
        /// @param path The path for the lock file
```

File: src/agent/include/cmd_ln_parser.hpp (hunk 7 above)
Comment (sdvendramini) [reply src/agent/include/cmd_ln_parser.hpp]: Done!

Comment (sdvendramini) [reply src/agent/include/cmd_ln_parser.hpp]: Done!

File: src/agent/src/unix_daemon.hpp (hunk 8 above)
Comment (sdvendramini) [reply src/agent/src/unix_daemon.hpp]: Done!

**Pull request merged**